pandas/_libs/lib.pyx

```diff
@@ -2845,15 +2845,17 @@ no_default = _NoDefault.no_default  # Sentinel indicating the default value.
 NoDefault = Literal[_NoDefault.no_default]
 
 
+@cython.boundscheck(False)
+@cython.wraparound(False)
 def map_infer_mask(
-        ndarray[object] arr,
-        object f,
-        const uint8_t[:] mask,
-        *,
-        bint convert=True,
-        object na_value=no_default,
-        cnp.dtype dtype=np.dtype(object)
-) -> "ArrayLike":
+    ndarray arr,
+    object f,
+    const uint8_t[:] mask,
+    *,
+    bint convert=True,
+    object na_value=no_default,
+    object dtype=np.dtype(object)
```

Comment:
why this change?  is the new numpy string dtype not an instance of cnp.dtype?

Reply:
I think this should be an accidental change I forgot to roll back.

In general, is there a reason we prefer ``cnp.dtype``?

With ``cnp.dtype`` I find it harder to do stuff like kind checks on the dtype object.